src/NanoManager.cpp

```diff
@@ -564,10 +571,14 @@ bool doDebuff(CNSocket *sock, sSkillResult_Buff *respdata, int i, int32_t target
 
     respdata[i].eCT = 4;
     respdata[i].iID = mob->appearanceData.iNPC_ID;
-    respdata[i].iConditionBitFlag = mob->appearanceData.iConditionBitFlag |= bitFlag;
-    mob->unbuffTimes[bitFlag] = getTime() + duration * 100;
-
-    std::cout << (int)mob->appearanceData.iNPC_ID << " was debuffed" << std::endl;
+    respdata[i].bProtected = 1;
+    if (mob->skillStyle < 0 && mob->state != MobState::RETREAT) { // only debuff if the enemy is not retreating and not casting corruption
+        mob->appearanceData.iConditionBitFlag |= bitFlag;
+        mob->unbuffTimes[bitFlag] = getTime() + duration * 100;
+        std::cout << (int)mob->appearanceData.iNPC_ID << " was debuffed" << std::endl;
```

Comment:
We can probably get rid of this print now, since we're trying to declutter the terminal.
```suggestion
```